remapping_commands.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+#requires vcf2bed, bowtie2, samtools, bedtools and reverse_strand.py (uses bamnostic)
+
+#This basic pipeline allows you to input two assemblies and a variant vcf file, 
+#and remap the variants to the new genome
+
+
+#example command:
+#time ./remapping_commands.sh -g droso_dm3.fasta -n GCA_000001215.4_Release_6_plus_ISO1_MT_genomic.fna -a GCA_000001215.4 -v droso_variants_renamed.vcf -o test.vcf
+
+
+#To be added:
+# -support for indels
+# -find a way to extract new assembly accession so it's not required as input? or just use new assembly file name?
+# -decide what to do in the case of new REF allele being the same as the variant allele (was A > G on old assembly, now is G > G on new assembly)
+
+#Completed:
+# -filter only SNPs (not indels) using bcftools filter
+# -copy filter etc columns from original vcf
+# -copy vcf header to the new vcf file, but regenerate contigs from the list of contigs in new assembly, and change reference tag
+# -take into account reverse strand
+# -create a command line version where you input file names
+# -% of remapped variants calculation
+
+
+oldgenome=''
+newgenome=''
+newgenomeaccession=''
+vcffile=''
+outfile=''
+
+while getopts 'g:n:a:v:o:' flag; do
+	case "${flag}" in
+		g) oldgenome="${OPTARG}";;
+		n) newgenome="${OPTARG}" ;;
+		a) newgenomeaccession="${OPTARG}" ;;
+		v) vcffile="${OPTARG}" ;;
+		o) outfile="${OPTARG}" ;;
+	esac
+done
+
+
+#1) Flanking sequence generation
+echo "-----------------------1) Flanking sequence generation-----------------------"
+#generate chrom sizes:
+samtools faidx $oldgenome 
+cut -f1,2 "$oldgenome".fai > "$oldgenome".chrom.sizes
+
+#filter SNPs only
+bcftools filter -i 'TYPE="snp"' $vcffile > snps_only.vcf
+
+#store header
+awk '$1 ~ /^#/ {print $0}' snps_only.vcf > vcf_header.txt
+
+#convert vcf to bed:
+vcf2bed < snps_only.vcf > variants.bed
+#the actual position of the variant is the second coordinate
+rm snps_only.vcf
+
+#generate the flanking sequence intervals
+bedtools flank -i variants.bed -g "$oldgenome".chrom.sizes -b 50 > flanking.bed
```

Comment:
Theoretically, even though unlikely, it is possible for a variant to be so close to the end of the chromosome that it's impossible to get a 50 bp flanking sequence for it. Could you please investigate what would happen in such a case when trying to run this command?

_This suggestion is low-priority_

Reply:
Bedtools flank requires a file with the length of each chromosome/contig (.chrom.sizes), this is used exactly for this purpose: 
[From the bedtools website:](https://bedtools.readthedocs.io/en/latest/content/tools/flank.html)
> Note that flank will restrict the created flanking intervals to the size of the chromosome (i.e. no start < 0 and no end > chromosome size).

So the read would be shorter, but no errors would occur.

Reply:
Right, but then this would cause problems later in the pipeline because we expect the SNP to always be at 51st position of the read; if the read is shorter the offset will be different. We should probably add a step where all reads shorter than the expected length (101) are filtered out prior to alignment

Reply:
(This remains low-priority)

Reply:
I added a filter: all reads not equal to 101 bases are now filtered out.